R/data-combined.R

```diff
@@ -0,0 +1,163 @@
+#' @title Object combining simulated and observed data
+#'
+#' @description
+#'
+#' A class for storage of simulated and/or observed data, which can be further
+#' used to extract as a dataframe or for visualization methods.
+#'
+#' @param simulationResults Instance of the `SimulationResults` object. A list
+#'   of such instances will **not** be accepted.
+#' @param dataSet Instance (or a list of instances) of the `DataSet` object(s).
+#' @param groups TODO:
+#' @param paths Quantity paths (element or vector of strings) for which the
+#'   results are to be returned. When providing the paths, only absolute full
+#'   paths are supported (i.e., no matching with '*' possible). If `NULL`
+#'   (default value), returns the results for all output defined in the results.
+#' @param individualIds A list of IDs of individuals whose simulations are of
+#'   interest.
+#'
+#' @examples
+#'
+#' # load the simulation
+#' simFilePath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
+#' sim <- loadSimulation(simFilePath)
+#' simulationResults <- runSimulations(simulations = sim)
+#'
+#' # create a new dataset object
+#' dataSet <- DataSet$new()
+#'
+#' # created object with datasets combined
+#' myCombDat <- DataCombined$new()
+#' myCombDat$addSimulationResults(simulationResults)
+#' myCombDat$addDataSet(dataSet)
+#'
+#' # print the object
+#' myCombDat
+#' @docType class
+#' @export
+
+DataCombined <- R6::R6Class(
+  classname = "DataCombined",
+  inherit = Printable,
+
+  # public fields and methods ------------------------------------
+
+  public = list(
+
+    ## constructor method ---------------
+
+    #' @description
+    #' Initialize a new instance of the class.
+    #' @return A new instance of `DataCombined` object.
+
+    # empty constructor because we prefer that the users add simulationResults
```

Comment:
Why do you prefer this? Can't you make them optional parameters at least?

Reply:
@PavelBal made a point that otherwise the signature of the constructor will grow to an unwieldy length because each individual method's parameters will need to be included.

I am fine either way.

Reply:
@msevestre I have made a proposal for the API of `DataCombined`, please review it here: https://github.com/Open-Systems-Pharmacology/OSPSuite-R/issues/674